src/Microsoft.Extensions.Configuration.AzureAppConfiguration/SrvLookupClient.cs

```diff
@@ -0,0 +1,134 @@
+
+            if (!exists || originHost.CacheExpires <= DateTimeOffset.UtcNow)
+            {
+                IEnumerable<SrvRecord> records = await InternalQueryAsync(originSrvDns, cancellationToken).ConfigureAwait(false);
+
+                if (records == null || records.Count() == 0)
+                {
+                    return Enumerable.Empty<SrvRecord>();
+                }
+
+                if (!exists)
+                {
+                    originHost = new OriginHostCacheItem()
+                    {
+                        OriginHost = records.First().Target.Value.TrimEnd('.'),
+                        CacheExpires = DateTimeOffset.UtcNow.Add(OriginHostResultCacheExpiration)
+                    };
+
+                    _cachedOriginHosts[originSrvDns] = originHost;
+                }
+                else
+                {
+                    originHost.OriginHost = records.First().Target.Value.TrimEnd('.');
+                    originHost.CacheExpires = DateTimeOffset.UtcNow.Add(OriginHostResultCacheExpiration);
+                }
+            }
+
+            int index = 0;
+
+            while (true)
+            {
+                string altSrvDns = $"{Alt}{index}.{TCP}.{originHost.OriginHost}";
+
+                IEnumerable<SrvRecord> records = await InternalQueryAsync(altSrvDns, cancellationToken).ConfigureAwait(false);
+
+                if (records == null)
+                {
+                    break;
+                }
+
+                resultRecords = resultRecords.Concat(records);
+
+                // If we get less than 20 records from _alt{i} SRV, we have reached the end of _alt* list
+                if (records.Count() < MaxSrvRecordCountPerRecordSet)
+                {
+                    break;
+                }
+
+                index++;
+            }
+
+            return resultRecords;
+        }
+
+        private async Task<IEnumerable<SrvRecord>> InternalQueryAsync(string srvDns, CancellationToken cancellationToken)
+        {
+            using CancellationTokenSource cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            cts.CancelAfter(UdpSrvQueryTimeout);
+
+            IDnsQueryResponse dnsResponse;
+
+            try
+            {
+                dnsResponse = await _udpLookupClient.QueryAsync(srvDns, QueryType.SRV, QueryClass.IN, cts.Token).ConfigureAwait(false);
+            }
+            catch (DnsResponseException) 
+            {
+                dnsResponse = await _tcpLookupClient.QueryAsync(srvDns, QueryType.SRV, QueryClass.IN, cancellationToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException) when (!cancellationToken.IsCancellationRequested)
+            {
+                dnsResponse = await _tcpLookupClient.QueryAsync(srvDns, QueryType.SRV, QueryClass.IN, cancellationToken).ConfigureAwait(false);
+            }
+
+            return dnsResponse.Answers.SrvRecords();

```

Comment:
What happens if there are no srv records for the store (store with no replicas). does `.SrvRecords()` return an empty enumerable?

Reply:
Yes, it returns an empty enumerable